server.test.js

```diff
@@ -66,12 +70,68 @@ describe('Suite of unit tests', function() {
                           "act": "join",
                           "isGenerator": "true"
                         });
+      genID = res.body.id;
       expect(res.body.ready).toEqual(true);
       expect(res.body.isGen).toEqual(true);
       expect(res.body.numPlayers).toEqual(2);
       expect(game.generatorID).toBeTruthy();
       done();
     });
 
+    it('should return bad response when malformed requests are given', async function(done) {
+      const badGen = await request(server)
+                        .post('/')
+                        .send({
+                          "act": "join",
+                          "isGenerator": ""
+                        });
+      expect(badGen.statusCode).toEqual(400);
+      done();
+    });
+  });
+
+  describe('setWordToGuess', function() {
+
+    it('should be able to set new word', async function(done) {
+      const res = await request(server)
+                        .post('/')
+                        .send({
+                          "act": "word",
+                          "word": "debug",
+                          "id": JSON.stringify(genID)
+                        });
+      expect(res.statusCode).toEqual(200);
+      expect(res.body.display).toEqual(['_','_','_','_','_']);
+      expect(res.body.guesses).toEqual(6);
+      expect(res.body.isOver).toEqual(false);
+      expect(game.wordToGuess).toEqual('debug');
+      done();
+    });
+
+    it('should return an error for malformed bodies', async function(done) {
+      const badWord = await request(server)
+                        .post('/')
+                        .send({
+                          "act": "word",
+                          "word": "",
+                          "id": JSON.stringify(guesserID)
+                        });
+      expect(badWord.statusCode).toEqual(400);
+
+      const badid = await request(server)
+                        .post('/')
+                        .send({
+                          "act": "word",
+                          "word": "asd",
+                          "id": JSON.stringify(guesserID)
+                        });
+      expect(badid.statusCode).toEqual(401);
+      expect(game.wordToGuess).toEqual('debug');
+      done();
+    });
+  });
```

Comment:
I've always wanted to put sad path testing into it's own describe block for visibility sake. Or maybe we could organize it some other way. What do you think? Got any ideas?

Reply:
Hmmm, interesting, I've never thought to do it that way. 

As I'm writing it at least, I feel when I'm writing tests for a single section, it feels easier to write the tests all at once. That way, I don't have to re-familiarize with the content and the ins and outs of what I'm dealing with. I feel like (in this case any way) that it's easier to procedurally go through the pieces and break them one by one to generate sad-paths. 

I'm very curious what doing it the other way might feel like!